psi4/src/psi4/libmints/matrix.cc

```diff
@@ -404,6 +404,43 @@ void Matrix::copy(const Matrix &cp) { copy(&cp); }
 
 void Matrix::copy(const SharedMatrix &cp) { copy(cp.get()); }
 
+// produces an Eigen::Map object mapping to the matrix data buffer 
+// of a matrix with a single irrep 
+Eigen::Map<Eigen::MatrixXd> Matrix::eigen_map() {
+    // this function only works with matrices with a single irrep 
+    if (nirrep() != 1) {
+        std::string message = "Matrix::eigen_map() called, but matrix only has one irrep! ";
+        message += "Use Matrix::eigen_maps() instead."; 
+        
+        throw PSIEXCEPTION(message); 
+    }
+
+    // create Eigen matrix "map" using Psi4 matrix data array directly
+    return std::move(
+        Eigen::Map<Eigen::MatrixXd>(
+            get_pointer(), nrow(), ncol()
+        )
+    );
+}
+
+// produces Eigen::Maps object mapping to the matrix data buffer
+// of a matrix with multiple irreps 
+// NOTE: this impl for mapping Psi4 matrices to Eigen maps
+// is currently experimental, as it is unused in the code 
+// currently
+std::vector<Eigen::Map<Eigen::MatrixXd>> Matrix::eigen_maps() {
+    std::vector<Eigen::Map<Eigen::MatrixXd>> eigen_maps;
+    eigen_maps.reserve(nirrep());
+
+    // create Eigen matrix "map"s for each irrep, 
+    // using Psi4 matrix data array directly
+    for (int h = 0; h != nirrep(); ++h) {
+        eigen_maps.emplace_back(get_pointer(h), rowdim(h), coldim(h));
+    } 
+    
+    return std::move(eigen_maps);
```

Comment:
Is this necessary?

Reply:
It's probably not "necessary", strictly speaking, but I did this on the understanding that doing a `std::move` on the output vector would be more efficient due to avoiding unnecessary object copying/duplication.

Reply:
Wouldn't return value optimization cover this?

Reply:
Yeah, when I double-checked my assumption online, it does seem that `std::move` can hinder RVO. So with that in mind, I'll get rid of the move.

Reply:
Done and done!